### DISCUSSION TITLE
'cancel' input function documentation (in 'bind' manpage)

### DISCUSSION THREAD

**Discussion by elder-n00b** [fish-shell/fish-shell#9644] [Q&A] [answered]
The fish-shell `bind` manpage documents a list of input functions, among which there is `cancel`:
~~~
cancel cancel the current commandline and replace it with a new empty one
~~~
Except it doesn't? It cancels a completion pager if there is one, but doesn't seem to affect the command line at all.
`kill-whole-line` does what the documentation says for `cancel`.
Am I missing something or is the documentation wrong?

---
~~~
fish❯ fish --version
fish, version 3.6.0
~~~

**Comment by zanchey** [comment, marked as answer]
Yes, the documentation is wrong. The source used to read "// The only thing we can cancel right now is paging, which we handled up above." Now it closes the pager and also undoes a recently-inserted completion.

**Comment by elder-n00b** [comment]
> also undoes a recently-inserted completion

Great!

I currently have `bind \e 'commandline -P; and commandline -f cancel; or commandline -f kill-whole-line'` in my config, based on the assumption that `cancel` only cancels the pager. It looks like I'm disabling the undo this way though. Thanks for the heads up!
